File: user-service/src/main/java/com/phorus/userservice/config/CustomProperties.java
package com.phorus.userservice.config;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration @Getter @Setter
@ConfigurationProperties(prefix = "com.phorus.userservice.services")
public class CustomProperties {

    private String petServiceUrl;
}

File: user-service/src/main/kotlin/com/phorus/userservice/config/CustomProperties.kt
package com.phorus.userservice.config;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration
@ConfigurationProperties(prefix = "com.phorus.userservice.services")
class CustomProperties(var petServiceUrl: String = "http://pet-service:8080")